packages/blocks/src/api/parser/index.js

```diff
@@ -101,6 +101,7 @@ export function normalizeRawBlock( rawBlock, options ) {
 	// meaning there are no negative consequences to repeated autop calls.
 	if (
 		rawBlockName === fallbackBlockName &&
+		rawBlockName !== 'core/html' &&
```

Comment:
This makes sense. I dare say this could even be `rawBlockName === 'core/freeform'` — to me, autop and freeform go together as descendants of WP's original simplified approach to post_content, where any block of text would be interpreted as a P tag. In other words, I can't think of a scenario with any block type other than freeform that would want autop.